app/controllers/ckeditor/pictures_controller.rb

```diff
@@ -0,0 +1,36 @@
+class Ckeditor::PicturesController < Ckeditor::ApplicationController
+  load_and_authorize_resource
+  def index
+    @pictures = Ckeditor.picture_adapter.find_all(ckeditor_pictures_scope)
+    @pictures = Ckeditor::Paginatable.new(@pictures).page(params[:page])
+
+    respond_to do |format|
+      format.html { render :layout => @pictures.first_page? }
+    end
+  end
+
+  def create
+    @picture = Ckeditor.picture_model.new
+    respond_with_asset(@picture)
+  end
+
+  def destroy
+    @picture.destroy
+
+    respond_to do |format|
+      format.html { redirect_to pictures_path }
+      format.json { render :nothing => true, :status => 204 }
+    end
+  end
+
+  protected
+
+    def find_asset
+      @picture = Ckeditor.picture_adapter.get!(params[:id])
+    end
+
+    def authorize_resource
+      model = (@picture || Ckeditor.picture_model)
```

Comment:
Are the parenthesis around `picture || Ckeditor.picture_model` necessary?

Reply:
@aitbw This file is almost completely copied from the `ckeditor` gem. That's why its code conventions have nothing to do with the rest of the project. IMHO, if it's possible, it'd be great if we didn't have to copy the file and could just add one or two changes to this class using something like the `CKEditor` initializer.